examples/metaflow-hf-trainer-card-callback/train.py

```diff
@@ -0,0 +1,101 @@
+import os
+import time
+import tarfile
+import argparse
+from random import choice
+from typing import Union, Dict
+
+from metaflow.plugins.hf_callbacks import DeepspeedHFTrainerAzureBlobSync
+
+from datasets import load_dataset
+from transformers import AutoModelForSequenceClassification
+from transformers import TrainingArguments
+from transformers import AutoTokenizer
+from transformers import DataCollatorWithPadding
+from transformers import Trainer
+from transformers import TrainerCallback
+
+try: # in case you want to download to local env without deepspeed
+    from deepspeed import comm as dist
+    from deepspeed import init_distributed
+except ImportError:
+    pass
+
+from metaflow.plugins.hf_info_card import MetaflowHuggingFaceCardCallback
+
+def main(
```

Comment:
Let's add an eval set so we have a more meaningful experiment tracking view on first UX:

```python
def main(
    checkpoint_dir: str = "training_outputs",
    ds_name: str = "rotten_tomatoes",
    model_name: str = "distilbert-base-uncased",
    train_split: str = "train[:50%]",
    eval_split: str = "validation",
    run_id: str = None,
    flow_name: str = None,
    local_rank: int = None,
    global_rank: int = None,
):
    train_ds = load_dataset(ds_name, split=train_split)
    eval_ds = load_dataset(ds_name, split=eval_split)
    tokenizer = AutoTokenizer.from_pretrained(model_name)
    model = AutoModelForSequenceClassification.from_pretrained(model_name)
    tokenize_dataset = lambda dataset: tokenizer(dataset["text"])
    train_ds = train_ds.map(tokenize_dataset, batched=True)
    eval_ds = eval_ds.map(tokenize_dataset, batched=True)

    data_collator = DataCollatorWithPadding(tokenizer=tokenizer)
    training_args = TrainingArguments(
        output_dir=checkpoint_dir,
        overwrite_output_dir=True,
        learning_rate=1e-5,
        do_train = True,
        do_eval = True,
        evaluation_strategy = "epoch",
        disable_tqdm = True, # this example logs progess bars to Metaflow GUI using MetaflowHuggingFaceCardCallback
        per_device_train_batch_size=8,
        per_device_eval_batch_size=8,
        num_train_epochs=5,
        logging_steps=1,
        # save 2 checkpoints (best one and last one)
        save_total_limit=0,
        save_steps=200,
        save_strategy="steps",
        load_best_model_at_end=False,
        save_on_each_node=False,  # default. Notice relation to DeepspeedHFTrainerS3Sync.push_from_all_nodes.
        warmup_steps=1,
        report_to="none",
    )
    trainer = Trainer(
        model=model,
        args=training_args,
        train_dataset=train_ds,
        eval_dataset=eval_ds,
        tokenizer=tokenizer,
        data_collator=data_collator,
        callbacks=[
            MetaflowHuggingFaceCardCallback(
                tracked_metrics=[
                    "loss",
                    "learning_rate",
                    "grad_norm",
                    "eval_loss"
                ]
            )
        ],
    )
    trainer.train()
```